### ISSUE TITLE
riak_moss_get_fsm:waiting_value/2 can get 'stop' event

### ISSUE THREAD

**Issue by slfritchie** [basho/riak_cs#236]
While meck testing, I think this is a legit timing bug?

```
00:28:29.861 [error] gen_fsm <0.12385.0> in state waiting_value terminated with reason: call to undefined function riak_moss_get_fsm:waiting_value/2 from gen_fsm:handle_msg/7 line 494
00:28:29.862 [error] CRASH REPORT Process <0.12385.0> with 0 neighbours exited with reason: call to undefined function riak_moss_get_fsm:waiting_value(stop, {state,undefined,<0.12386.0>,<0.12384.0>,<<"testMECK">>,#Ref<0.0.0.40839>,<<"foo">>,[],undefined,...}) in gen_fsm:terminate/7 line 611
```

**Comment by reiddraper** [comment]
Any hints on reproducing? Is this from calling the public `stop` fun, or is the FSM sending the stop event to itself when it's in the `waiting_value` state?

**Comment by slfritchie** [comment]
That's a good question.  I managed to do it by using Meck to fake Riak responses, and perhaps doing one or more in less than one millisecond is enough.  I confess that I haven't looked at it.

Here's the patch that I've used to fix:

```
diff --git a/src/riak_moss_get_fsm.erl b/src/riak_moss_get_fsm.erl
index a77303a..8616cbb 100644
--- a/src/riak_moss_get_fsm.erl
+++ b/src/riak_moss_get_fsm.erl
@@ -33,6 +33,7 @@
 -export([init/1,
          prepare/2,
          prepare/3,
+         waiting_value/2,
          waiting_value/3,
          waiting_continue_or_stop/2,
          waiting_chunks/2,
@@ -151,6 +152,9 @@ prepare(get_manifest, _From, State) ->
             {reply, Mfst, waiting_continue_or_stop, NewState, NextStateTimeout}
     end.

+waiting_value(stop, State) ->
+    {stop, normal, State}.
+
 waiting_value(get_manifest, _From, State=#state{manifest=undefined}) ->
     {stop, normal, notfound, State};
 waiting_value(get_manifest, _From, State=#state{manifest=Mfst}) ->
```

**Comment by slfritchie** [comment]
Silly me for not just fixing this, duh.  https://github.com/basho/riak_cs/pull/307

**Comment by slfritchie** [comment]
Closed by #307
